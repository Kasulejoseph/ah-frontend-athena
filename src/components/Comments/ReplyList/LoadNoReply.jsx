import React from 'react';
import PropTypes from 'prop-types';
import { MDBCard, MDBCardBody, MDBContainer, MDBBadge } from 'mdbreact';
import CommentEdit from '../CommentEdit/CommentEdit';
import Reply from './ReplyList';
import ReplyBox from '../../../views/CommentView/ReplyBox/ReplyBox';

const LoadNoReply = ({
  comment,
  id,
  clickEdit,
  clickDelete,
  clickReply,
  parentId,
  isEdit,
  childId,
  showComBox,
  username,
  clickLike,
  clickDisLike
}) => {
  return (
    <div>
      <MDBContainer fluid className="mw-75 ml-5 p-0 mx-1 my-3">
        <MDBCard className="mw-75 p-0 m-0">
          <MDBCardBody className="mw-75 m-0 p-2 ">
            <h6 className="comment-body mb-0 col-12">{comment.comment_body}</h6>
            <div className="col-3 user-info  offset-9 border border-white">
              <div className="date">
                <MDBBadge color="light" className="mr-2">
                  {comment.dateHour}
                </MDBBadge>
                Hours ago
              </div>
              <div className="mdc-chip">
                <img
                  src={`${comment.author.image}`}
                  className="img-fluid z-depth-1 square mr-2  rounded-circle"
                  alt="Contact Person"
                />
                {username}
              </div>
            </div>
          </MDBCardBody>
          <CommentEdit
            likes={comment.likes_count}
            clickEdit={clickEdit}
            clickLike={clickLike}
            clickDisLike={clickDisLike}
            clickDelete={clickDelete}
            id={id}
            clickReply={clickReply}
          />
          <Reply parentId={id} replyList={{}} />
          <div className={showComBox}>
            <ReplyBox isEdit={isEdit} parentId={parentId} childId={childId} />
          </div>
        </MDBCard>
      </MDBContainer>
    </div>
  );
};

LoadNoReply.propTypes = {
  comment: PropTypes.shape({}),
  id: PropTypes.number.isRequired,
  clickEdit: PropTypes.func.isRequired,
  clickDelete: PropTypes.func.isRequired,
  clickReply: PropTypes.func.isRequired,
  parentId: PropTypes.number,
  isEdit: PropTypes.bool.isRequired,
  childId: PropTypes.number,
  showComBox: PropTypes.func.isRequired,
  username: PropTypes.string,
  clickLike: PropTypes.func.isRequired,
  clickDisLike: PropTypes.func.isRequired
};

LoadNoReply.defaultProps = {
  comment: {},
  parentId: 1,
  childId: 1,
  username: ''
};

export default LoadNoReply;
