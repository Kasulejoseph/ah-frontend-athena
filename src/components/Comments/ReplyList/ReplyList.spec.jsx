import React from 'react';
import { shallow } from 'enzyme';
import ReplyBoxList, { ReplyList, mapStateToProps } from './ReplyList';
import LoadNoReply from './LoadNoReply';

let props = {
  commentList: [],
  showReplies: 'd-none',
  showComBox: 'd-none',
  isEdit: false,
  clickDelete: jest.fn(),
  clickLike: jest.fn(),
  clickDisLike: jest.fn(),
  CommentDislikeAction: jest.fn(),
  CommentLikeAction: jest.fn()
};
const commentData = {
  articles: { view_article: { slug: 'mocked' } },
  replyList: {
    replies: [
      {
        comment_body: 'my comment',
        author: { image: 'url' }
      }
    ]
  }
};

describe('ReplyList test', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<ReplyList {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle clickReply function sets showcomBox to d-none', () => {
    const wrapper = shallow(<ReplyList {...props} />);
    wrapper.setState({ showComBox: '' });
    wrapper.instance().clickReply();
    expect(wrapper.state('showComBox')).toEqual('d-none');
  });
  it('should handle clickReply function sets showcomBox to none', () => {
    const wrapper = shallow(<ReplyList {...props} />);
    wrapper.instance().clickReply();
    expect(wrapper.state('showComBox')).toEqual('');
  });

  it('should handle clickLike', () => {
    const wrapper = shallow(<ReplyBoxList {...props} />);
    wrapper.setState({ showReplies: 'd-none' });
    wrapper.simulate('onClick');
    wrapper.instance().props.clickLike(1);
    wrapper.instance().props.CommentLikeAction(1, '');

    expect(wrapper.state('showReplies')).toEqual('d-none');
  });

  it('should handle clickDisLike', () => {
    const wrapper = shallow(<ReplyBoxList {...props} />);
    wrapper.setState({ showReplies: 'd-none' });
    wrapper.instance().props.clickDisLike(1);

    expect(wrapper.state('showReplies')).toEqual('d-none');
  });

  it('should handle clickDelete', () => {
    const wrapper = shallow(<ReplyBoxList {...props} />);
    wrapper.setState({ showReplies: 'd-none' });
    wrapper.instance().props.clickDelete(1);

    expect(wrapper.state('showReplies')).toEqual('d-none');
  });

  it('should handle clickEdit and set edit to true', () => {
    const wrapper = shallow(<ReplyList {...props} />);
    wrapper.setState({ showComBox: '' });
    wrapper.instance().clickEdit(1);
    expect(wrapper.state('showComBox')).toEqual('d-none');
  });

  it('should handle clickEdit when showcomBox is set to d-none', () => {
    const wrapper = shallow(<ReplyList {...props} />);
    wrapper.setState({ showReplies: '' });
    wrapper.instance().clickEdit(1);
    expect(wrapper.state('showReplies')).toEqual('');
  });

  it('should handle repliesShow and set repliesShow to d-none', () => {
    const wrapper = shallow(<ReplyList {...props} />);
    wrapper.setState({ showReplies: '' });
    wrapper.instance().repliesShow();
    expect(wrapper.state('showReplies')).toEqual('d-none');
  });

  it('should handle repliesShow and set repliesShow to none', () => {
    const wrapper = shallow(<ReplyList {...props} />);
    wrapper.setState({ showReplies: 'd-none' });
    wrapper.instance().repliesShow();
    expect(wrapper.state('showReplies')).toEqual('');
  });

  it('should handle dateToHours', () => {
    const wrapper = shallow(<ReplyList {...props} />);
    wrapper.instance().dateToHours();
    expect(wrapper.state('isEdit')).toEqual(false);
  });

  it('should will recieve reply props', () => {
    const wrapper = shallow(<ReplyBoxList {...props} />);
    wrapper.setProps({ commentData });
    expect(wrapper.instance().props.commentData).toEqual(commentData);
  });

  it('should map state to props', () => {
    expect(mapStateToProps(commentData)).toEqual({ slug: 'mocked' });
  });

  it('snapshot LoadNoReply dump component', () => {
    const image = { comment: { author: { image: 'url' } } };
    const wrapper = shallow(<LoadNoReply {...image} />);
    expect(wrapper).toMatchSnapshot();
  });
});
