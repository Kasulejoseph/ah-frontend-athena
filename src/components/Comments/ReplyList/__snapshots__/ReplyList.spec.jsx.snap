// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReplyList test matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReplyList
    CommentDislikeAction={[MockFunction]}
    CommentLikeAction={[MockFunction]}
    clickDelete={[MockFunction]}
    clickDisLike={[MockFunction]}
    clickLike={[MockFunction]}
    commentList={Array []}
    isEdit={false}
    parentId={null}
    replyList={Object {}}
    show={[Function]}
    showComBox="d-none"
    showReplies="d-none"
    slug=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [],
        <ModalPage
          fallback="/"
          md="12"
          modal={false}
          title="Please first login "
          toggle={[Function]}
        />,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "fallback": "/",
          "history": undefined,
          "md": "12",
          "modal": false,
          "title": "Please first login ",
          "toggle": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [],
          <ModalPage
            fallback="/"
            md="12"
            modal={false}
            title="Please first login "
            toggle={[Function]}
          />,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "fallback": "/",
            "history": undefined,
            "md": "12",
            "modal": false,
            "title": "Please first login ",
            "toggle": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`ReplyList test snapshot LoadNoReply dump component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoadNoReply
    childId={1}
    comment={
      Object {
        "author": Object {
          "image": "url",
        },
      }
    }
    parentId={1}
    username=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <t
        className="mw-75 ml-5 p-0 mx-1 my-3"
        fluid={true}
        tag="div"
      >
        <t
          className="mw-75 p-0 m-0"
          tag="div"
        >
          <t
            className="mw-75 m-0 p-2 "
            tag="div"
          >
            <h6
              className="comment-body mb-0 col-12"
            />
            <div
              className="col-3 user-info  offset-9 border border-white"
            >
              <div
                className="date"
              >
                <t
                  className="mr-2"
                  color="light"
                  pill={false}
                  tag="span"
                />
                Hours ago
              </div>
              <div
                className="mdc-chip"
              >
                <img
                  alt="Contact Person"
                  className="img-fluid z-depth-1 square mr-2  rounded-circle"
                  src="url"
                />
                
              </div>
            </div>
          </t>
          <CommentEdit />
          <Connect(ReplyList)
            replyList={Object {}}
          />
          <div>
            <Connect(ReplyBox)
              childId={1}
              parentId={1}
            />
          </div>
        </t>
      </t>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <t
          className="mw-75 p-0 m-0"
          tag="div"
        >
          <t
            className="mw-75 m-0 p-2 "
            tag="div"
          >
            <h6
              className="comment-body mb-0 col-12"
            />
            <div
              className="col-3 user-info  offset-9 border border-white"
            >
              <div
                className="date"
              >
                <t
                  className="mr-2"
                  color="light"
                  pill={false}
                  tag="span"
                />
                Hours ago
              </div>
              <div
                className="mdc-chip"
              >
                <img
                  alt="Contact Person"
                  className="img-fluid z-depth-1 square mr-2  rounded-circle"
                  src="url"
                />
                
              </div>
            </div>
          </t>
          <CommentEdit />
          <Connect(ReplyList)
            replyList={Object {}}
          />
          <div>
            <Connect(ReplyBox)
              childId={1}
              parentId={1}
            />
          </div>
        </t>,
        "className": "mw-75 ml-5 p-0 mx-1 my-3",
        "fluid": true,
        "tag": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <t
              className="mw-75 m-0 p-2 "
              tag="div"
            >
              <h6
                className="comment-body mb-0 col-12"
              />
              <div
                className="col-3 user-info  offset-9 border border-white"
              >
                <div
                  className="date"
                >
                  <t
                    className="mr-2"
                    color="light"
                    pill={false}
                    tag="span"
                  />
                  Hours ago
                </div>
                <div
                  className="mdc-chip"
                >
                  <img
                    alt="Contact Person"
                    className="img-fluid z-depth-1 square mr-2  rounded-circle"
                    src="url"
                  />
                  
                </div>
              </div>
            </t>,
            <CommentEdit />,
            <Connect(ReplyList)
              replyList={Object {}}
            />,
            <div>
              <Connect(ReplyBox)
                childId={1}
                parentId={1}
              />
            </div>,
          ],
          "className": "mw-75 p-0 m-0",
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <h6
                  className="comment-body mb-0 col-12"
                />,
                <div
                  className="col-3 user-info  offset-9 border border-white"
                >
                  <div
                    className="date"
                  >
                    <t
                      className="mr-2"
                      color="light"
                      pill={false}
                      tag="span"
                    />
                    Hours ago
                  </div>
                  <div
                    className="mdc-chip"
                  >
                    <img
                      alt="Contact Person"
                      className="img-fluid z-depth-1 square mr-2  rounded-circle"
                      src="url"
                    />
                    
                  </div>
                </div>,
              ],
              "className": "mw-75 m-0 p-2 ",
              "tag": "div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "comment-body mb-0 col-12",
                },
                "ref": null,
                "rendered": null,
                "type": "h6",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="date"
                    >
                      <t
                        className="mr-2"
                        color="light"
                        pill={false}
                        tag="span"
                      />
                      Hours ago
                    </div>,
                    <div
                      className="mdc-chip"
                    >
                      <img
                        alt="Contact Person"
                        className="img-fluid z-depth-1 square mr-2  rounded-circle"
                        src="url"
                      />
                      
                    </div>,
                  ],
                  "className": "col-3 user-info  offset-9 border border-white",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <t
                          className="mr-2"
                          color="light"
                          pill={false}
                          tag="span"
                        />,
                        "Hours ago",
                      ],
                      "className": "date",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": undefined,
                          "className": "mr-2",
                          "color": "light",
                          "pill": false,
                          "tag": "span",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      "Hours ago",
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <img
                          alt="Contact Person"
                          className="img-fluid z-depth-1 square mr-2  rounded-circle"
                          src="url"
                        />,
                        "",
                      ],
                      "className": "mdc-chip",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "alt": "Contact Person",
                          "className": "img-fluid z-depth-1 square mr-2  rounded-circle",
                          "src": "url",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "img",
                      },
                      "",
                    ],
                    "type": "div",
                  },
                ],
                "type": "div",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "clickDelete": undefined,
              "clickDisLike": undefined,
              "clickEdit": undefined,
              "clickLike": undefined,
              "clickReply": undefined,
              "id": undefined,
              "likes": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "parentId": undefined,
              "replyList": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Connect(ReplyBox)
                childId={1}
                parentId={1}
              />,
              "className": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "childId": 1,
                "isEdit": undefined,
                "parentId": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <t
          className="mw-75 ml-5 p-0 mx-1 my-3"
          fluid={true}
          tag="div"
        >
          <t
            className="mw-75 p-0 m-0"
            tag="div"
          >
            <t
              className="mw-75 m-0 p-2 "
              tag="div"
            >
              <h6
                className="comment-body mb-0 col-12"
              />
              <div
                className="col-3 user-info  offset-9 border border-white"
              >
                <div
                  className="date"
                >
                  <t
                    className="mr-2"
                    color="light"
                    pill={false}
                    tag="span"
                  />
                  Hours ago
                </div>
                <div
                  className="mdc-chip"
                >
                  <img
                    alt="Contact Person"
                    className="img-fluid z-depth-1 square mr-2  rounded-circle"
                    src="url"
                  />
                  
                </div>
              </div>
            </t>
            <CommentEdit />
            <Connect(ReplyList)
              replyList={Object {}}
            />
            <div>
              <Connect(ReplyBox)
                childId={1}
                parentId={1}
              />
            </div>
          </t>
        </t>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <t
            className="mw-75 p-0 m-0"
            tag="div"
          >
            <t
              className="mw-75 m-0 p-2 "
              tag="div"
            >
              <h6
                className="comment-body mb-0 col-12"
              />
              <div
                className="col-3 user-info  offset-9 border border-white"
              >
                <div
                  className="date"
                >
                  <t
                    className="mr-2"
                    color="light"
                    pill={false}
                    tag="span"
                  />
                  Hours ago
                </div>
                <div
                  className="mdc-chip"
                >
                  <img
                    alt="Contact Person"
                    className="img-fluid z-depth-1 square mr-2  rounded-circle"
                    src="url"
                  />
                  
                </div>
              </div>
            </t>
            <CommentEdit />
            <Connect(ReplyList)
              replyList={Object {}}
            />
            <div>
              <Connect(ReplyBox)
                childId={1}
                parentId={1}
              />
            </div>
          </t>,
          "className": "mw-75 ml-5 p-0 mx-1 my-3",
          "fluid": true,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <t
                className="mw-75 m-0 p-2 "
                tag="div"
              >
                <h6
                  className="comment-body mb-0 col-12"
                />
                <div
                  className="col-3 user-info  offset-9 border border-white"
                >
                  <div
                    className="date"
                  >
                    <t
                      className="mr-2"
                      color="light"
                      pill={false}
                      tag="span"
                    />
                    Hours ago
                  </div>
                  <div
                    className="mdc-chip"
                  >
                    <img
                      alt="Contact Person"
                      className="img-fluid z-depth-1 square mr-2  rounded-circle"
                      src="url"
                    />
                    
                  </div>
                </div>
              </t>,
              <CommentEdit />,
              <Connect(ReplyList)
                replyList={Object {}}
              />,
              <div>
                <Connect(ReplyBox)
                  childId={1}
                  parentId={1}
                />
              </div>,
            ],
            "className": "mw-75 p-0 m-0",
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <h6
                    className="comment-body mb-0 col-12"
                  />,
                  <div
                    className="col-3 user-info  offset-9 border border-white"
                  >
                    <div
                      className="date"
                    >
                      <t
                        className="mr-2"
                        color="light"
                        pill={false}
                        tag="span"
                      />
                      Hours ago
                    </div>
                    <div
                      className="mdc-chip"
                    >
                      <img
                        alt="Contact Person"
                        className="img-fluid z-depth-1 square mr-2  rounded-circle"
                        src="url"
                      />
                      
                    </div>
                  </div>,
                ],
                "className": "mw-75 m-0 p-2 ",
                "tag": "div",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "comment-body mb-0 col-12",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h6",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <div
                        className="date"
                      >
                        <t
                          className="mr-2"
                          color="light"
                          pill={false}
                          tag="span"
                        />
                        Hours ago
                      </div>,
                      <div
                        className="mdc-chip"
                      >
                        <img
                          alt="Contact Person"
                          className="img-fluid z-depth-1 square mr-2  rounded-circle"
                          src="url"
                        />
                        
                      </div>,
                    ],
                    "className": "col-3 user-info  offset-9 border border-white",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <t
                            className="mr-2"
                            color="light"
                            pill={false}
                            tag="span"
                          />,
                          "Hours ago",
                        ],
                        "className": "date",
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "class",
                          "props": Object {
                            "children": undefined,
                            "className": "mr-2",
                            "color": "light",
                            "pill": false,
                            "tag": "span",
                          },
                          "ref": null,
                          "rendered": null,
                          "type": [Function],
                        },
                        "Hours ago",
                      ],
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <img
                            alt="Contact Person"
                            className="img-fluid z-depth-1 square mr-2  rounded-circle"
                            src="url"
                          />,
                          "",
                        ],
                        "className": "mdc-chip",
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "alt": "Contact Person",
                            "className": "img-fluid z-depth-1 square mr-2  rounded-circle",
                            "src": "url",
                          },
                          "ref": null,
                          "rendered": null,
                          "type": "img",
                        },
                        "",
                      ],
                      "type": "div",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "clickDelete": undefined,
                "clickDisLike": undefined,
                "clickEdit": undefined,
                "clickLike": undefined,
                "clickReply": undefined,
                "id": undefined,
                "likes": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "parentId": undefined,
                "replyList": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(ReplyBox)
                  childId={1}
                  parentId={1}
                />,
                "className": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "childId": 1,
                  "isEdit": undefined,
                  "parentId": 1,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
